workflow/controller/controller.go

```diff
@@ -767,7 +767,13 @@ func (wfc *WorkflowController) processNextItem(ctx context.Context) bool {
 		return true
 	}
 	startTime := time.Now()
-	woc.operate(ctx)
+
+	// check if workflow is already completed
+	if !woc.wf.Status.Fulfilled() {
```

Comment:
I think the right fix is here. Workflow should not operate if workflow is already fulfilled

Reply:
I recommend doing this inside `operate`. We will still want the first deferred function to run

```go
func (woc *wfOperationCtx) operate(ctx context.Context) {
	defer func() {
		if woc.wf.Status.Fulfilled() {
			woc.killDaemonedChildren("")
		}
		woc.persistUpdates(ctx)
	}()

        // ...

	// Do not operate on Fulfilled workflows
	if woc.wf.Fulfilled() {
		return
	}

        // ...
}

Reply:
make sense. This will enable code testability.